rustls/src/client/tls12.rs

```diff
@@ -755,8 +755,8 @@ impl State<ClientConnectionData> for ExpectServerDone {
         // 4.
         if let Some(client_auth) = &st.client_auth {
             let certs = match client_auth {
-                ClientAuthDetails::Empty { .. } => Vec::new(),
-                ClientAuthDetails::Verify { certkey, .. } => certkey.cert.clone(),
+                ClientAuthDetails::Empty { .. } => <_>::default(),
```

Comment:
Let's just write `CertificateChain::default()` here?

Reply:
for what it's worth, rust-analyzer's autocompletion suggests `Default::default()` in scenarios like "functional update syntax" (shown below)
``` rust
let new = Struct {
    some_field: new_value,
    .. Default::default() // <- suggestion upon typing `.`
};
```

would that be acceptable? PR #1597 is going to use that syntax a lot so being able to use auto-completion (without having to manually patch it up to adhere to style conventions) would certainly help

Reply:
That form seems reasonable to me 👍

Reply:
> I usually associate "standalone" as a free-standing fn outside of an impl block, but that's the terminology the [Rust language docs for the impl keyword](https://doc.rust-lang.org/std/keyword.impl.html) use:

IME, *traits aside*, `Foo::constructor()` that returns `Foo` (or a type that contain `Foo` like `Option<Foo>` or `Result<Foo, _>`) is called a "constructor". `Foo::function()` that does not meet the preceding criteria is called "an associated function" (just like `Foo::CONSTANT` is "associated constant"; everything inside an `impl` block is an associated item). `default` is a member of the `Default` trait so I would probably call it a "trait function" because there's an orthogonal axis: an "inherent method" is a method in `impl Foo`; a "trait method" is a method in  `impl Trait for Foo`. 

I guess a catch all could be to say "do not elide names when calling a function using [fully-qualified syntax](https://doc.rust-lang.org/beta/reference/expressions/call-expr.html#disambiguating-function-calls)" (when I learned that syntax it was still called UFCS)

Reply:
Thanks for the clarification, that's helpful. I'll let djc weigh in and then consider a CONTRIB update PR.

Reply:
I also prefer to avoid `Default::default()` for the simple reason that for `Type::default()` IMO the benefits (local clarity, especially when outside an IDE context, for example during review) outweigh the costs (potentially a type name that's slightly more verbose or maybe an extra import).

I agree that it would make for a good style guide addition to say that fully-qualified function calls should not elide the type.

Reply:
> I agree that it would make for a good style guide addition to say that fully-qualified function calls should not elide the type.

https://github.com/rustls/rustls/pull/1620